src/DnsResolver/custom/csharp/RestErrorHandler.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Management.Automation;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Azure.PowerShell.Cmdlets.DnsResolver.Models.Api20220701;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.PowerShell.Cmdlets.DnsResolver.Cmdlets
+{
+    internal static class CmdletRestExtension
+    {
+        public static void WriteError(this Cmdlet cmdlet, HttpResponseMessage responseMessage, Task<ICloudError> errorResponseTask, ref Task<bool> returnNow)
+        {
+            var errorResponse = errorResponseTask?.ConfigureAwait(false).GetAwaiter().GetResult();
+
+            if (errorResponse?.Detail != null)
+            {
+                var errorDetail = errorResponse.Detail;
+                var errorDetailString = JsonConvert.SerializeObject(errorDetail);
```

Comment:
Do not use newtonsoft, which is not included in autogen modules. Autorest.powershell actually has an embedded Json support. And [here ](https://github.com/Azure/azure-powershell/blob/main/src/Databricks/generated/api/Models/Api20210401Preview/ErrorInfo.json.cs#L108-L112)is an example about how to use it. Besides, you may loop the errorDetail to call errorDetail[i].ToJsonString(), and then combine them together. Actually, I think you may just need to combine errorDetail[i].Message, and do not need to return the target and the code.